state/factory/workingset.go

```diff
@@ -170,6 +171,7 @@ func (ws *workingSet) runAction(
 	}
 	for _, actionHandler := range reg.All() {
 		receipt, err := actionHandler.Handle(ctx, elp.Action(), ws)
+		ws.clearActFunc()
```

Comment:
clear all snapshots made by current tx

Reply:
```
 	elpHash, err := elp.Hash()
	if err != nil {
		return nil, errors.Wrapf(err, "Failed to get hash")
	}

	var receipt *action.Receipt
	var err error
	for _, actionHandler := range reg.All() {
		receipt, err = actionHandler.Handle(ctx, elp.Action(), ws)
		if err != nil {
			err = errors.Wrapf(
				err,
				"error when action %x mutates states",
				elpHash,
			)
			break
		}
		if receipt != nil {
			break
		}
	}
	ws.clearActFunc()
	// TODO (zhi): return error if both receipt and error are nil
	return receipt, err
```

Reply:
good improvement. this will save lock time